src/plugins/mf_priority.cpp

```diff
@@ -198,12 +198,181 @@ int check_queue_factor (flux_plugin_t *p,
 }
 
 
+/*
+ * Add held job IDs to a JSON array to be added to a bank_info JSON object.
+ */
+static json_t *add_held_jobs (
+                            const std::pair<std::string, struct bank_info> &b)
+{
+    json_t *held_jobs = NULL;
+
+    // add any held jobs to a JSON array
+    if (!(held_jobs = json_array ()))
+        goto error;
+
+    for (auto const &j : b.second.held_jobs) {
+        json_t *jobid = json_integer (j);
+
+        if (!jobid)
+            goto error;
+
+        if (json_array_append (held_jobs, jobid) < 0) {
+            json_decref (jobid);
+            goto error;
+        }
```

Comment:
Use `json_array_append_new()` here.  The distinction is that `_new()` doesn't take a reference on `jobid`.  It "steals" the existing reference, so once that call is successful, you no longer have to worry about freeing `jobid`.  Conversely, when it fails you do.

Reply:
I see. Thanks for pointing this out! There are two other calls to `json_array_append ()` in the helper functions (in `banks_to_json ()` and `query_cb ()`, but I left those as is because I'm pretty sure I free the JSON object after the call to `array_append ()`. Let me know if those would be better to also change to `array_append_new ()`.